File: task-app/frontend/src/types/task.ts
export type TaskStatus   = 'todo' | 'in-progress' | 'done';
export type TaskPriority = 'low' | 'medium' | 'high';

export interface Task {
  id: number;
  title: string;
  description: string;
  status: TaskStatus;
  priority: TaskPriority;
  category: string;
  due_date: string | null;
  created_at: string;
  updated_at: string;
}

export interface CreateTaskInput {
  title: string;
  description?: string;
  status?: TaskStatus;
  priority?: TaskPriority;
  category?: string;
  due_date?: string | null;
}

export interface TaskFilters {
  status?: TaskStatus | '';
  priority?: TaskPriority | '';
  category?: string;
  q?: string;
}

export interface TaskStats {
  total: number;
  todo: number;
  in_progress: number;
  done: number;
}

File: taskflow-android/app/src/main/java/com/taskflow/android/data/Task.kt
package com.taskflow.android.data

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.time.LocalDateTime

@Entity(tableName = "tasks")
data class Task(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val title: String,
    val description: String = "",
    val dueDate: LocalDateTime? = null,
    val priority: Priority = Priority.MEDIUM,
    val status: Status = Status.TODO,
    val isHabit: Boolean = false,
    val recurrence: Recurrence? = null,
    val createdAt: LocalDateTime = LocalDateTime.now(),
    val completedAt: LocalDateTime? = null,
)

enum class Priority { LOW, MEDIUM, HIGH }
enum class Status   { TODO, IN_PROGRESS, DONE }
enum class Recurrence { DAILY, WEEKLY, BIWEEKLY, MONTHLY, CUSTOM }
